components/FlexibleHeader/src/private/MDCStatusBarShifter.m

```diff
@@ -150,6 +152,14 @@ - (void)attemptSnapshotState:(MDCStatusBarShifterState)snapshotState {
     snapshotState = MDCStatusBarShifterStateInvalidSnapshot;
   }
 
+  // Invalidate the snapshot if our replica view is currently hidden and we're attempting to take
+  // a new snapshot. This handles the case where you're running on an iPhone X in landscape, you
+  // hide the header, and then rotate back to portrait. It is at this point that we want to
+  // invalidate the snapshot.
+  if (_isChangingInterfaceOrientation && snapshotState == MDCStatusBarShifterStateIsSnapshot) {
```

Comment:
Maybe move this and the above `if` statement into one.

`if (snapshotState == MDCStatusBarShifterStateIsSnapshot) {
      if ((_isChangingInterfaceOrientation) || (!_snapshottingEnabled && _snapshotState == MDCStatusBarShifterStateRealStatusBar)) {
           snapshotState = MDCStatusBarShifterStateInvalidSnapshot;
      }
}`

Possibly makes it more readable.

Reply:
Ah I see what you mean. My intent with this block of code is to explicitly document each of the state machine's edges, so they are intentionally separate.